airbyte-metrics/metrics-lib/src/main/java/io/airbyte/metrics/lib/OpenTelemetryMetricClient.java

```diff
@@ -75,9 +80,33 @@ public void initialize(MetricEmittingApp metricEmittingApp, String otelEndpoint)
                 .build())
         .setResource(resource)
         .build();
-    OtlpGrpcMetricExporter metricExporter = OtlpGrpcMetricExporter.builder()
+    metricExporter = OtlpGrpcMetricExporter.builder()
         .setEndpoint(otelEndpoint).build();
-    SdkMeterProvider meterProvider = SdkMeterProvider.builder()
+    initialize(metricEmittingApp, sdkTracerProvider, resource);
+  }
+
+  @VisibleForTesting
+  void initializeWithInMemoryExporter(MetricEmittingApp metricEmittingApp) {
```

Comment:
if this is only for testing, it is worth moving this out and placing it in the test class?